### PR TITLE
Add CAPI predicates to RKEBootstrap

### PR DESCRIPTION
**Pull request by jakefhyde**
## Issue: 

Part of groundwork for #39139

## Problem

We need a mechanism to support pausing the various CRs in our CAPI implementation, so we can disable cluster reconciliation during disruptive actions such as encryption key rotation.

## Solution

Add missing predicates to rke bootstrap controller, including checking for pausing as well as other required predicates we were missing.

https://cluster-api.sigs.k8s.io/developer/providers/bootstrap.html?highlight=bootstrap#bootstrap-provider-specification

**Note:** there are some predicates which have been added, which do not appear in the upstream docs as requirements, however these predicates (for example, checking if `capiCluster.Status.InfrastructureReady == false`) are added because although we functionally always set `InfrastructureReady = true`, bootstrapping before the cluster is ready is not desired and is subject to change in the future.

### CODE REVIEW

File: pkg/controllers/provisioningv2/rke2/rkebootstrap/controller.go [hunk 1]
```diff
@@ -1,10 +1,10 @@
-package bootstrap
```
Comment (Oats87) [inline pkg/controllers/provisioningv2/rke2/rkebootstrap/controller.go:1]: I would prefer we do not rename packages at this time. It makes maintenance very difficult. We'll be much better off doing this once we've stabilized our code base.

**Review by Oats87 — CHANGES_REQUESTED** [review]

File: pkg/controllers/provisioningv2/rke2/bootstrap/controller.go [hunk 2]
```diff
@@ -276,31 +272,58 @@ func (h *handler) OnChange(bootstrap *rkev1.RKEBootstrap, status rkev1.RKEBootst
 		result []runtime.Object
 	)
 
+	machine, err := rke2.GetOwnerCAPIMachine(bootstrap, h.machineCache)
+	if apierrors.IsNotFound(err) {
```
Comment (annablender) [inline pkg/controllers/provisioningv2/rke2/bootstrap/controller.go:276]: This is fine the way you have it. I feel like it's more `Go style` for error handling but personally, I'd put the `IsNotFound` checks inside the `err != nil` blocks. That makes more logical sense but again I see the efficiency of short circuiting here

**Review by annablender — DISMISSED** [review]

**Review by HarrisonWAffel — DISMISSED** [review]
This looks good and matches the conditions within the CAPI docs

File: pkg/controllers/provisioningv2/rke2/bootstrap/controller.go [hunk 3]
```diff
@@ -20,12 +20,15 @@ import (
 	"github.com/rancher/wrangler/pkg/generic"
 	"github.com/rancher/wrangler/pkg/name"
 	"github.com/rancher/wrangler/pkg/relatedresource"
+	"github.com/sirupsen/logrus"
 	corev1 "k8s.io/api/core/v1"
 	rbacv1 "k8s.io/api/rbac/v1"
 	apierror "k8s.io/apimachinery/pkg/api/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
```
Comment (Oats87) [inline pkg/controllers/provisioningv2/rke2/bootstrap/controller.go:27]: this is imported already

**Review by Oats87 — CHANGES_REQUESTED** [review]

**Review by HarrisonWAffel — APPROVED** [review]

**Review by HarrisonWAffel — APPROVED** [review]

**Review by Oats87 — APPROVED** [review]
LGTM!

**Pull request merged**